test/include/dlaf_test/matrix/matrix_local.h

```diff
@@ -0,0 +1,159 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2020, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <type_traits>
+
+#include "dlaf/common/range2d.h"
+#include "dlaf/matrix/layout_info.h"
+#include "dlaf/matrix/tile.h"
+#include "dlaf/memory/memory_view.h"
+#include "dlaf/types.h"
+#include "dlaf/util_math.h"
+
+namespace dlaf {
+namespace matrix {
+namespace test {
+
+template <class T>
+struct MatrixLocal;
+
+/// MatrixLocal is a local matrix with column-layout, not thread-safe.
+///
+/// It is a useful helper object that allows you to access elements directly given their
+/// GlobalElementIndex, or by accessing the tile containing it via its GlobalTileIndex and then the
+/// related LocalTileIndex inside it.
+///
+/// It uses Index/Size with the Global tag instead of the Local one, because its main task
+/// is to create a local copy of a distributed matrix. So, it is generally easier to think
+/// of it as the global matrix.
+///
+/// About its semantic, it is similar to a std::unique_ptr.
```

Comment:
I would add that that it uses column major layout and the full matrix can be used in blas and lapack.

Reply:
I agree about BLAS/LAPACK, I'm going to add it.

About the column major layout I'm interested in knowing if **"MatrixLocal is a local matrix with column-layout, not thread-safe"** is ok like this or you have interpreted it differently.

Reply:
Please note that the correct wording is "column-major layout".
With "local matrix with column-layout" I understand a matrix with m >> n (like a col panel).
As brief doc I would use `MatrixLocal is a column-major local matrix, not thread-safe.`
In the doc body I would connect the column-major layout to the possibility to use it in blas and lapack, and expand also the meaning of thread-safe (i.e. no futures used => cannot be used to schedule tasks)

Reply:
Do you think is it more clear now? Do you think it would be better to change/add anything else?

Reply:
LGTM